portal/views/portal.py

```diff
@@ -696,18 +696,32 @@ def settings():
     return response
 
 
+@portal.route('/api/settings', defaults={'config_key': None})
 @portal.route('/api/settings/<string:config_key>')
 @oauth.require_oauth()
 def config_settings(config_key):
-    key = config_key.upper()
-    # handing out LifeRay keys at this time
-    if key.startswith('LR_'):
-        return jsonify({key: current_app.config.get(key)})
-    # handing out consent related keys e.g. consent_with_top_level_org
-    elif key.startswith('CONSENT_'):
-        return jsonify({key: current_app.config.get(key)})
+    if config_key:
+        key = config_key.upper()
+        # handing out LifeRay keys at this time
+        if key.startswith('LR_'):
+            return jsonify({key: current_app.config.get(key)})
+        # handing out consent related keys e.g. consent_with_top_level_org
+        elif key.startswith('CONSENT_'):
```

Comment:
The next time you have to add another prefix/key, you might try something like this:
```
config_prefix_whitelist = (
    'REQUIRED_CORE_DATA',
    'SYSTEM',
)
if not any(key.startswith(prefix) for prefix in config_prefix_whitelist):
    abort(400, "Configuration key '{}' not available".format(key))
return jsonify({key: current_app.config.get(key)})
```